src/lib/y2users/linux/base_reader.rb

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "abstract_method"
+require "y2users/config"
+require "y2users/parsers/group"
+require "y2users/parsers/passwd"
+require "y2users/parsers/shadow"
+require "users/ssh_authorized_keyring"
+
+module Y2Users
+  module Linux
+    # Base class for reading users configuration from the system
+    class BaseReader
+      include Yast::Logger
+
+      # Generates a new config with the users and groups from the read content
+      #
+      # @return [Config]
+      def read
+        elements = read_users + read_groups
+
+        config = Config.new.attach(elements)
+
+        # read passwords after user, as user has to exist in advance
+        read_passwords(config)
+
+        # read authorized keys
```

Comment:
Do we need a comment saying "read authorized keys" to explain the line of code `read_authorized_keys(c)`?

Reply:
Not at all